### ISSUE TITLE
REST get categories for store "all" missing data

### ISSUE THREAD

**Issue by materi** [magento/magento2#25269]
REST get categories for store "all" missing data (/rest/all/V1/categories).
Important properties like `name`,`is_active` are null.
Using a specific store, for example "default" still works and we get working data.

I have not tried to reproduce the issue on fresh install, noticed after upgrade from 2.3.x to 2.3.3.

### Preconditions (*)

1. Magento 2.3.3 CE

### Steps to reproduce (*)

1. Call GET /rest/all/V1/categories; `curl -H "Authorization: Bearer MYTOKEN" -H "Accept: application/json" https://mymagentoshop/rest/all/V1/categories`

### Expected result (*)
```json
{
  "id": 1,
  "parent_id": 0,
  "name": "Root Catalog",
  "is_active": null,
  "position": 0,
  "level": 0,
  "product_count": 0,
  "children_data": [
    {
      "id": 2,
      "parent_id": 1,
      "name": "Default Category",
      "is_active": true,
      "position": 1,
      "level": 1,
      "product_count": 0,
      "children_data": [
        {
          "id": 278,
          "parent_id": 2,
          "name": "My Category A",
          "is_active": true,
          "position": 16,
          "level": 2,
          "product_count": 100,
          "children_data": []
        },
        {
          "id": 309,
          "parent_id": 2,
          "name": "My Category B",
          "is_active": true,
          "position": 23,
          "level": 2,
          "product_count": 200,
          "children_data": []
        }
      ]
    }
  ]
}
```

### Actual result (*)
```json
{
  "id": 1,
  "parent_id": 0,
  "name": null,
  "is_active": null,
  "position": 0,
  "level": 0,
  "product_count": null,
  "children_data": [
    {
      "id": 2,
      "parent_id": 1,
      "name": null,
      "is_active": null,
      "position": 1,
      "level": 1,
      "product_count": null,
      "children_data": [
        {
          "id": 278,
          "parent_id": 2,
          "name": null,
          "is_active": null,
          "position": 16,
          "level": 2,
          "product_count": null,
          "children_data": []
        },
        {
          "id": 309,
          "parent_id": 2,
          "name": null,
          "is_active": null,
          "position": 23,
          "level": 2,
          "product_count": null,
          "children_data": []
        }
      ]
    }
  ]
}
```

**Comment by Stepa4man** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @Stepa4man. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @Stepa4man, here is your [Magento instance](https://i-25269-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-25269-2-3-develop.instances.magento-community.engineering/admin](https://i-25269-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by kanthy-magento** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @itsPranith. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by engcom-Charlie** [comment]
Hello @materi 

Thank you for contribution and collaboration!

We are not able to reproduce this issue on the latest 2.3-develop branch(after upgrade from 2.3.2).

**Manual testing scenario:**

1. Upgrade from 2.3.2 to 2.3-develop
2. Call `GET /rest/all/V1/categories`

**Actual result:**
Response as expected:
![image](https://user-images.githubusercontent.com/51680850/67580271-bc4a3380-f74e-11e9-8478-a479dd939566.png)

So this issue should be closed.
Thanks for your report!

**Comment by siliconalchemy** [comment]
I confirm exactly the same issue in Magento 2.3.3.  This is a killer when you depend on retrieving all categories across all stores to sync with any other system.
@engcom-Charlie can you please reopen?

**Comment by siliconalchemy** [comment]
Ah, this has since been fixed:
https://github.com/magento/magento2/commit/eb28168184ff769e0f03c8c2e5c2fe223cdac7a2#diff-06419995ef6a4b327e3306cfba805293R110

The bug is in line 110 in  app/code/Magento/Catalog/Model/Category/Tree.php:
Change:
        `$this->categoryTree->addCollectionData($this->categoryCollection);`
to:
        `$categoryTree->addCollectionData($this->categoryCollection);`

I assume this will be released in the next 2.3 point release?

**Comment by engcom-Charlie** [comment]
Hello @siliconalchemy 
Yes, fix will be included in 2.3.4 :)

**Comment by siliconalchemy** [comment]
Fab, thanks :)
